crates/pixi_build_backend_passthrough/src/lib.rs

```diff
@@ -122,13 +169,80 @@ impl InMemoryBackend for PassthroughBackend {
     }
 }
 
+/// Creates a conda package on the fly with the given IndexJson metadata.
+fn create_conda_package_on_the_fly(
+    index_json: &IndexJson,
+    output_path: &Path,
+) -> Result<(), std::io::Error> {
+    use rattler_conda_types::compression_level::CompressionLevel;
+
+    // Create a temporary directory to stage the package contents
+    let temp_dir = tempfile::tempdir()?;
+    let info_dir = temp_dir.path().join("info");
+    fs_err::create_dir_all(&info_dir)?;
+
+    // Write index.json
+    let index_json_content = serde_json::to_string_pretty(index_json)?;
```

Comment:
When you create the outputs from conda_outputs you augment the information from the index.json. That should also happen here, otherwise they dont represent the same thing.